x-pack/plugins/actions/server/builtin_action_types/email.ts

```diff
@@ -136,12 +146,12 @@ export function getActionType(params: GetActionTypeParams): ActionType {
 
 async function executor(
   { logger }: { logger: Logger },
-  execOptions: ActionTypeExecutorOptions
+  execOptions: EmailActionTypeExecutorOptions
 ): Promise<ActionTypeExecutorResult> {
```

Comment:
we're missing out on a chance to use the generic Data parameter type in the ActionTypeExecutorResult type - and I think most of the other action types.  We could do this as a follow-on.  Primary use case is to help ensure action authors are returning the data they are claiming they want to return

Reply:
Good point, I went ahead and added support for this in this commit: https://github.com/elastic/kibana/pull/71439/commits/ab64713fb294edc95b0d35cbb047c5bdf1e4e06a.

Turns out all executors return some form of `ActionTypeExecutorResult`. Some with and some without `data` so I was able to make it more explicit.

Reply:
ya, looks good, thx!